### ISSUE TITLE
Can't get WiVRn to work on Steam Deck.

### ISSUE THREAD

**Issue by RaymondYouTube** [WiVRn/WiVRn#158]
Okay so I have WiVRn all set up on my Steam Deck. I'm using a Meta Quest 3. I booted up Quake VR, a VR mod for Quake 1 made by vittorioromeo, the game works, but it's not detecting the headset. I tried Lutris, I tried the Heroic Games Launcher, and I even tried Steam using this "PRESSURE_VESSEL_FILESYSTEMS_RW=$XDG_RUNTIME_DIR/wivrn/comp_ipc:/var/lib/flatpak/app/io.github.wivrn.wivrn %command%" launch option. No dice. I've been told that the performance of WiVRn is better than ALVR via SteamVR, but I wouldn't really know because WiVRn is not detecting the headset. The Steam Deck uses SteamOS, which is based on Arch Linux. I know the Steam Deck does support VR headsets technically, because it works with ALVR. So, does anyone know how to fix this issue? If anyone knows, please let me know. Thank you. :)

**Comment by xytovl** [comment]
I doubt Steam Deck is powerful enough for any VR content, but should still work.

If your headset can't detect the server (shows message "Start a WiVRn server on your local network"), make sure you have followed steps in https://github.com/WiVRn/WiVRn?tab=readme-ov-file#my-computer-is-not-seen-by-the-headset

**Comment by RaymondYouTube** [comment]
I mean it did connect successfully, but the screen on the headset is stuck on "Connection ready. Start a VR application on steamdeck". And with Quake VR like I mentioned earlier boots up but not showing on the headset. And I even checked the firewalls, and on my Steam Deck, the terminal says that opening ports for firewalls have already been enabled. Here's what it says "Warning: ALREADY_ENABLED: '9757:udp' already in 'public'". And yes, I have it running with "systemctl status avahi-daemon". So I tried another application, Beat Saber, and I couldn't get that to work either.

**Comment by galister** [comment]
does hello xr work?

**Comment by RaymondYouTube** [comment]
Is that part of OpenXR-SDK-Source or something?

**Comment by galister** [comment]
it ships with SteamVR, in its bin folder

Launch with `./helloxr_vulkan -v -g Vulkan2`

**Comment by RaymondYouTube** [comment]
In the launch options, right? Like this: "PRESSURE_VESSEL_FILESYSTEMS_RW=$XDG_RUNTIME_DIR/wivrn/comp_ipc:/var/lib/flatpak/app/io.github.wivrn.wivrn %command% ./helloxr_vulkan -v -g Vulkan2"

**Comment by galister** [comment]
nor sure what you're trying to do there. it's simply shipped as a binary file with steamvr. it's not possible to launch it from inside steam.

when wivrn says to launch an XR app:
```
cd ~/.steam/root/steamapps/common/SteamVR/bin/linux64
./helloxr_vulkan -v -g Vulkan2
```

**Comment by RaymondYouTube** [comment]
I see. Thanks.

**Comment by RaymondYouTube** [comment]
Okay, I tried it and it gave me this. "./helloxr_vulkan: error while loading shared libraries: libopenxr_loader.so: cannot open shared object file: No such file or directory"

**Comment by galister** [comment]
is this with the WiVRn dashboard, Envision, or just running from the command line?

**Comment by RaymondYouTube** [comment]
I typed it into the terminal.

**Comment by galister** [comment]
how are you starting wivrn?

**Comment by RaymondYouTube** [comment]
I type this in "systemctl enable --now avahi-daemon" before starting it up. I then connect my Meta Quest 3 to my Steam Deck using WiVRn. Now as for typing "cd ~/.steam/root/steamapps/common/SteamVR/bin/linux64" and "./helloxr_vulkan -v -g Vulkan2" in the terminal, I kept the WiVRn dashboard open.

**Comment by galister** [comment]
please show the output of 
`ls -la ~/.config/openxr/1/`
and
`cat ~/.config/openxr/1/active_runtime.json`
and also `ls -la` the `library_path` that's in the json file

**Comment by xytovl** [comment]
Do you still have the issue or have you been able to fix? If fixed please share the resolution/workaround

**Comment by RaymondYouTube** [comment]
> Do you still have the issue or have you been able to fix? If fixed please share the resolution/workaround

Sorry. I got distracted with other things. My bad. Anyway, to answer your question, no I still haven’t got WiVRn to work on my Steam Deck.

**Comment by xytovl** [comment]
> Okay, I tried it and it gave me this. "./helloxr_vulkan: error while loading shared libraries: libopenxr_loader.so: cannot open shared object file: No such file or directory"

This errors is caused by missing openxr loader on the system, on standard Arch Linux it would be provided by https://archlinux.org/packages/extra/x86_64/openxr/, on Steam Deck I don't know if it can be installed.

An useful debug method is to add `PROTON_LOG=1 WINEDEBUG=vrclient,openxr,steam` in the launch command for the Steam game you want to start.
This will write a log file in the home folder which should contain relevant information.

**Comment by RaymondYouTube** [comment]
> > Okay, I tried it and it gave me this. "./helloxr_vulkan: error while loading shared libraries: libopenxr_loader.so: cannot open shared object file: No such file or directory"
> 
> This errors is caused by missing openxr loader on the system, on standard Arch Linux it would be provided by https://archlinux.org/packages/extra/x86_64/openxr/, on Steam Deck I don't know if it can be installed.
> 
> An useful debug method is to add `PROTON_LOG=1 WINEDEBUG=vrclient,openxr,steam` in the launch command for the Steam game you want to start. This will write a log file in the home folder which should contain relevant information.

But what about VR games that have a native Linux version? Do I still need to type that into the launch options?

**Comment by xytovl** [comment]
Those options are to get relevant logs and understand why it fails.

I don't know if there are debug options for native games, VR often works better using proton anyway.

**Comment by RaymondYouTube** [comment]
Okay so, one of my buddies at the NightSightProductions discord server, King Vulpes, told me that he tried out WiVRn on the Steam Deck with Beat Saber, it worked for him. He is working on those videos on the NightSightProductions YouTube channel where he makes videos for the Steam Deck, AMD and Intel GPUs, and Apple hardware.

I will close this issue for now, as he is making a video on how to setup WiVRn on Steam Deck. If I encounter another issue, I will post it on my separate issue post.
